internal/services/compute/disk_encryption_set_resource.go

```diff
@@ -253,26 +320,50 @@ func resourceDiskEncryptionSetUpdate(d *pluginsdk.ResourceData, meta interface{}
 		update.Tags = tags.Expand(d.Get("tags").(map[string]interface{}))
 	}
 
+	rotationToLatestKeyVersionEnabled := d.Get("auto_key_rotation_enabled").(bool)
+	keyVaultKey, err := keyVaultParse.ParseOptionallyVersionedNestedItemID(d.Get("key_vault_key_id").(string))
+	if err != nil {
+		return err
+	}
+
 	if d.HasChange("key_vault_key_id") {
-		keyVaultKeyId := d.Get("key_vault_key_id").(string)
-		keyVaultDetails, err := diskEncryptionSetRetrieveKeyVault(ctx, keyVaultsClient, resourcesClient, keyVaultKeyId)
+		keyVaultDetails, err := diskEncryptionSetRetrieveKeyVault(ctx, keyVaultsClient, resourcesClient, keyVaultKey.ID())
 		if err != nil {
-			return fmt.Errorf("validating Key Vault Key %q for Disk Encryption Set: %+v", keyVaultKeyId, err)
+			return fmt.Errorf("validating Key Vault Key %q for Disk Encryption Set: %+v", keyVaultKey.ID(), err)
```

Comment:
Now we have this as an object, rather than a string, we should pass it in and skip having to `ID()` to a string just to have to re-parse it?